src/mono/netcore/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using NSSearchPathDirectory = Interop.Sys.NSSearchPathDirectory;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static Dictionary<SpecialFolder, string> s_specialFolders;
+
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            if (s_specialFolders == null)
+            {
+                Interlocked.CompareExchange(ref s_specialFolders, new Dictionary<SpecialFolder, string>(), null);
+            }
```

Comment:
nit: I think it's fine to initialize the field directly so we don't need the Interlocked.

Reply:
without CompareExchange it can allocate the dictionary twice

Reply:
I meant doing:

```
private static Dictionary<SpecialFolder, string> s_specialFolders = new Dictionary<SpecialFolder, string>();
```

Reply:
@akoeplinger yeah but it will add a static ctor, it looks like `Environment` still doesn't have one for performance reasons. There is another similiar code to init a dictionary in a similiar manner already, see `EnsureEnvironmentCached`

Reply:
This looks good